### PR TITLE
Deduplicate code for text-to-path conversion in svg backend.

### PR DESCRIPTION
**Pull request by anntzer**
... between the non-mathtext and mathtext cases.

The only slightly tricky point is whether we still need to special-case
the `len(glyph_path[0]) == 0` case; we don't, because self._convert_path
uses _path.convert_to_string, which handles zero-length paths just fine.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_svg.py [hunk 1]
```diff
@@ -912,86 +922,46 @@ def _draw_text_as_path(self, gc, x, y, s, prop, angle, ismath, mtext=None):
...
             style['opacity'] = short_float_fmt(alpha)
+        font_scale = fontsize / text2path.FONT_SCALE
+        attrib = {
+            'style': generate_css(style),
+            'transform': generate_transform([
+                ('translate', (x, y)),
+                ('rotate', (-angle,)),
+                ('scale', (font_scale, -font_scale))]),
+        }
+        writer.start('g', attrib=attrib)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_svg.py]: Is this the same element as previously l.944? If so

- why is it pulled out out of the `if not ismath`?
- AFAICS the `devs` element created in `_update_glyph_map_defs` is now a child of `g`, which it had not been before.

File: lib/matplotlib/backends/backend_svg.py [hunk 2]
```diff
@@ -912,86 +922,46 @@ def _draw_text_as_path(self, gc, x, y, s, prop, angle, ismath, mtext=None):
...
 
         if not ismath:
             font = text2path._get_font(prop)
             _glyphs = text2path.get_glyphs_with_font(
                 font, s, glyph_map=glyph_map, return_new_glyphs_only=True)
             glyph_info, glyph_map_new, rects = _glyphs
+            self._update_glyph_map_defs(glyph_map_new)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_svg.py]: This does `char_id = self._adjust_char_id(char_id)`, which the original code did not. I assume that was an oversight in the original code?

Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: This was an oversight, because svg ids should not contain "%20" (https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/id), which is the only thing _adjust_char_id replaces.  Likely the oversight was benign, because I doubt any non-mathtext glyph names contained that substring, but it doesn't hurt to run them through _adjust_char_id, as well.

File: lib/matplotlib/backends/backend_svg.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: a) to just have a single (begin-group, end-group) in the program flow.
b) per https://developer.mozilla.org/en-US/docs/Web/SVG/Element/defs https://www.w3.org/TR/SVG2/struct.html#Head https://www.w3.org/TR/SVGTiny12/struct.html#DefsElement we can put \<def> anywhere -- it's suggested to collect them all to the top of the file, but we don't even do that (that can't really work with the current design as we're streaming out the svg "on the fly").

File: lib/matplotlib/backends/backend_svg.py [hunk 3]
```diff
@@ -877,27 +877,37 @@ def draw_image(self, gc, x, y, im, transform=None):
...
+        writer = self.writer
+        if glyph_map_new:
+            writer.start('defs')
+            for char_id, (codes, vertices) in glyph_map_new.items():
+                char_id = self._adjust_char_id(char_id)
+                path_data = self._convert_path(
+                    Path(codes, vertices), simplify=False)
+                writer.element('path', id=char_id, d=path_data)
+            writer.end('defs')
+        self._glyph_map.update(glyph_map_new)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_svg.py]: In the original code, the updated was within `if glyph_map_new:`. Is this change intentional?

Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: Yes: update()ing a dict with an empty dict doesn't change it, so we can always do it.

Comment (timhoffm) [reply lib/matplotlib/backends/backend_svg.py]: But why should we? IMHO it's clearer to not call update if we know it does not have any effect. Any here, the code is not even getting more complex for the distinction.

Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: I think it's simpler to have less code under the branch, but I don't really care enough.  Moved it into the branch.

File: lib/matplotlib/backends/backend_svg.py [hunk 4]
```diff
@@ -877,27 +877,37 @@ def draw_image(self, gc, x, y, im, transform=None):
...
+        Emit definitions for not-yet-defined glyphs, and record them as having
+        been defined.
+        """
+        writer = self.writer
+        if glyph_map_new:
+            writer.start('defs')
+            for char_id, (codes, vertices) in glyph_map_new.items():
+                char_id = self._adjust_char_id(char_id)
+                path_data = self._convert_path(
+                    Path(codes, vertices), simplify=False)
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_svg.py]: and
```suggestion
                    Path(vertices, codes), simplify=False)
```

**Comment by anntzer** [comment]
oops, fixed
